### PR TITLE
submin a string

### PR DESCRIPTION
**Pull request by nikosnick13**
This is a test for my projects

### CODE REVIEW

**Comment by nikosnick13** [comment]
project finished

File: nikosnick13.HabitLogger/Program.cs [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+            bool isAppRunning = false;
+
+            while (!isAppRunning)
+            {
+               
+                WriteLine("\n\nMAIN Menu");
+                WriteLine("\nWhat would you like to do?");
+                WriteLine("\n\tType 0 to Close Application.");
+                WriteLine("\tType 1 to Inser a Record ");
```
Comment (chrisjamiecarter) [inline nikosnick13.HabitLogger/Program.cs:59]: Typo:
`WriteLine("\tType 1 to Inser a Record ");`

`WriteLine("\tType 1 to Insert a Record ");`

File: nikosnick13.HabitLogger/Program.cs [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        }
+
+        #region Insert
+
+        private static void Insert()
+        {
+            Clear();
+            string date = GetDateInput();
+
+            int quantity = GetNumberInput("\n\nPlaese Insert a number of glasses or another measures of your choice (no decimals is allowed \n\n) ");
```
Comment (chrisjamiecarter) [inline nikosnick13.HabitLogger/Program.cs:102]: Weird bracket on different line.

`int quantity = GetNumberInput("\n\nPlaese Insert a number of glasses or another measures of your choice (no decimals is allowed \n\n) ");`

`int quantity = GetNumberInput("\n\nPlaese Insert a number of glasses or another measures of your choice (no decimals is allowed) \n\n ");`

File: nikosnick13.HabitLogger/Program.cs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                insertUserCmd.CommandText = $"INSERT INTO dring_water(date,quantity ) VALUES ('{date}',{quantity})";
+                insertUserCmd.ExecuteNonQuery();
+                conn.Close();
+            }
+
+        }
+        //Get the input for Date as string
+        private static string GetDateInput()
+        {
+            WriteLine("Please insert tha date: dd-MM-yyyy. Type 0 to return to main menu.");
```
Comment (chrisjamiecarter) [inline nikosnick13.HabitLogger/Program.cs:118]: Typo

`WriteLine("Please insert tha date: dd-MM-yyyy. Type 0 to return to main menu.");`

`WriteLine("Please insert the date: dd-MM-yyyy. Type 0 to return to main menu.");`

**Review by chrisjamiecarter — CHANGES_REQUESTED** [review]
Hey @nikosnick13 ,

Excellent work on another application! 🎉 It’s great to see your progress.

I have performed my review, and you have failed to meet one requirement:

- [ ] Your project needs to contain a Read Me file where you'll explain how your app works.

Here are some additional suggestions that I believe can help you grow as a developer (they do not need actioning if you don't want to, the readme is the priority):

**Solution Structure**
So close on the solution structure! 
It should be `[ProjectName].[GitHubUserName]/` not `[GitHubUserName].[ProjectName]/`.

**Style Guide Compliance**
Your code is generally well-structured, but to align with best practices, it would be beneficial to follow [Microsoft's style guide](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions) more closely.
This will ensure consistency and make your code easier for others to read and maintain.
- Pay attention to the whitespace in your code. You have too much whitespace in your code. To make it more readable, one blank line should be the max between methods/statements/etc.
- Use file-scoped namespaces to save an indentation level:
```
// Block-Scoped Namespace.
namespace YourNamespace
{
    // Code here.
}

// vs

// File-Scoped Namespace
namespace YourNamespace;
// Code here.
```
- Use the simpler "Using Declaration" syntax to save an indentation level against the "Using Statement" syntax:
```
// Using Statement.
using (var connection = new SqliteConnection(connectionString))
{
    // Code here.
}

// vs

// Using Declaration
using var connection = new SqliteConnection(connectionString);
// Code here.
```

**SQL Injection Attacks**
Read this to prevent [SQL Injection Attacks](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html)
Basically, don't do:
`editCmd.CommandText = $"UPDATE dring_water SET Date = '{updateDate}',Quantity = {updareQuantity} WHERE Id = {grapId}";`
Do:
`editCmd.CommandText = $"UPDATE dring_water SET Date = @UpdateDate, Quantity = @UpdateQuantity WHERE Id = @Id";`
`editCmd.Parameters.AddWithValue("@UpdateDate", UpdateDate);`
`editCmd.Parameters.AddWithValue("@UpdateQuantity", UpdateQuantity);`
`editCmd.Parameters.AddWithValue("@Id", Id);`

**User Experience**
Try always leave a clean menu at the top of the console.
If you present information to the user that they need to read/digest, you can await an input to continue.
Then clear the console and re-write the menu.
This would allow the user to perform a "transaction" in their mind, maintain readability of the console window and improve the experience for the user.

Feel free to reach out if you have any questions or if you'd like to collaborate further on any of these points.

Please update to meet all the requirements, Commit and Push your updates. I will then re-review!

Thanks,
@chrisjamiecarter

File: nikosnick13.HabitLogger/Program.cs (hunk 2 above)
Comment (chrisjamiecarter) [reply nikosnick13.HabitLogger/Program.cs:102]: Also typo:
`PLaese`

`Please`

**Comment by nikosnick13** [comment]
Ok Chris i will check it and let you know !!!
Thanks for your mentoring !!!
________________________________

**Comment by nikosnick13** [comment]
#I fix the bugs
#Use Parameters - Using Declaration
#Correct spelling mistakes
#Add a README.md file

**Review by chrisjamiecarter — APPROVED** [review]
Hey @nikosnick13,

Great job on fulfilling the requirement!

I will go ahead and mark as approved, keep up the excellent work on the next projects! 😊

Best regards,
@chrisjamiecarter

**Pull request closed without merging**